server/controllers/thread.js

```diff
@@ -0,0 +1,42 @@
+const { Thread, Post } = require("../database");
+const config = require("../config/config");
+
+module.exports = {
+  createRequest: async data => {
+    const { title, content, language, user } = data;
+    const newPost = new Post({
+      author: user._id,
+      authorName: user.name,
+      data: content
+    });
+
+    const newThread = new Thread({
+      creator: user._id,
+      title,
+      status: 0,
+      language: { name: language.name, experience: language.experience }
+    });
+
+    const post = await newPost.save();
+    newThread.posts.push(post);
+    newThread.no_assign.push(user._id);
+    const resultThread = await newThread.save();
+    return resultThread;
+  },
+
+  addToNoAssign: async (threadId, userId) => {
+    const newThread = await Thread.findById(threadId);
+    newThread.no_assign.addToSet(userId);
+    const resultThread = await newThread.save();
+    return resultThread;
+  },
+
+  updateStatus: async (threadId, newStatus) => {
+    const status = config.server.threadStatus.indexOf(newStatus);
```

Comment:
Does this gracefully handle errors?

Reply:
For now I've going with wrapping any calls to controllers in catch/try blocks for error handling. It's consistent that way across the app so far.